sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2201,9 +2202,20 @@ object GpuOverrides {
     exec[BatchScanExec](
       "The backend for most file input",
       ExecChecks(
-        (TypeSig.commonCudfTypes + TypeSig.STRUCT + TypeSig.MAP + TypeSig.ARRAY).nested(),
+        (TypeSig.commonCudfTypes + TypeSig.STRUCT + TypeSig.MAP + TypeSig.ARRAY +
+            TypeSig.DECIMAL).nested(),
         TypeSig.all),
       (p, conf, parent, r) => new SparkPlanMeta[BatchScanExec](p, conf, parent, r) {
+        override def tagPlanForGpu(): Unit = {
+          val hasDecimal = p.scan.readSchema().exists(_.dataType.isInstanceOf[DecimalType])
+          if (hasDecimal) {
+            if (!(p.scan.isInstanceOf[ParquetScan] && this.conf.decimalTypeEnabled)) {
+              this.willNotWorkOnGpu(
+                "Only supports reading decimal from parquet with DECIMAL_TYPE_ENABLED = true")
```

Comment:
Users won't know what `DECIMAL_TYPE_ENABLED` is.  I don't think the decimal enable config is relevant here, and this should be simplified to something like, "Decimals are only supported by Parquet".

Reply:
Yes! I've removed the redundant `decimalTypeEnabled` config check.